From 71887b32f3bf6f5532d26c7a091555b3c67d9f64 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 8 Aug 2024 08:26:56 +1100
Subject: [PATCH 07/13] don't hide faiss::gpu symbols on windows

This is a windows-only patch; the python bindings need the GPU object
library linkage to propagate.

We also need to mark the symbols PUBLIC that faiss_gpu needs itself, e.g.
from cuBLAS.
---
 faiss/gpu/CMakeLists.txt    | 10 +++++-----
 faiss/python/CMakeLists.txt | 10 +++++-----
 2 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/faiss/gpu/CMakeLists.txt b/faiss/gpu/CMakeLists.txt
index 4dfdaf5..2ef652d 100644
--- a/faiss/gpu/CMakeLists.txt
+++ b/faiss/gpu/CMakeLists.txt
@@ -324,11 +324,11 @@ endif()
 # Export FAISS_GPU_HEADERS variable to parent scope.
 set(FAISS_GPU_HEADERS ${FAISS_GPU_HEADERS} PARENT_SCOPE)
 
-target_link_libraries(faiss PRIVATE  faiss_gpu_objs)
-target_link_libraries(faiss_avx2 PRIVATE faiss_gpu_objs)
-target_link_libraries(faiss_avx512 PRIVATE faiss_gpu_objs)
-target_link_libraries(faiss_avx512_spr PRIVATE faiss_gpu_objs)
-target_link_libraries(faiss_sve PRIVATE faiss_gpu_objs)
+target_link_libraries(faiss PUBLIC  faiss_gpu_objs)
+target_link_libraries(faiss_avx2 PUBLIC faiss_gpu_objs)
+target_link_libraries(faiss_avx512 PUBLIC faiss_gpu_objs)
+target_link_libraries(faiss_avx512_spr PUBLIC faiss_gpu_objs)
+target_link_libraries(faiss_sve PUBLIC faiss_gpu_objs)
 
 install(TARGETS faiss_gpu_objs EXPORT faiss-targets)
 
diff --git a/faiss/python/CMakeLists.txt b/faiss/python/CMakeLists.txt
index 1361113..9e4b085 100644
--- a/faiss/python/CMakeLists.txt
+++ b/faiss/python/CMakeLists.txt
@@ -237,11 +237,11 @@ if(FAISS_ENABLE_GPU)
     if(FAISS_ENABLE_CUVS AND NOT TARGET cuvs::cuvs)
       find_package(cuvs)
     endif()
-    target_link_libraries(swigfaiss PRIVATE CUDA::cudart $<$<BOOL:${FAISS_ENABLE_CUVS}>:cuvs::cuvs>)
-    target_link_libraries(swigfaiss_avx2 PRIVATE CUDA::cudart $<$<BOOL:${FAISS_ENABLE_CUVS}>:cuvs::cuvs>)
-    target_link_libraries(swigfaiss_avx512 PRIVATE CUDA::cudart $<$<BOOL:${FAISS_ENABLE_CUVS}>:cuvs::cuvs>)
-    target_link_libraries(swigfaiss_avx512_spr PRIVATE CUDA::cudart $<$<BOOL:${FAISS_ENABLE_CUVS}>:cuvs::cuvs>)
-    target_link_libraries(swigfaiss_sve PRIVATE CUDA::cudart $<$<BOOL:${FAISS_ENABLE_CUVS}>:cuvs::cuvs>)
+    target_link_libraries(swigfaiss PRIVATE CUDA::cudart CUDA::cublas $<$<BOOL:${FAISS_ENABLE_CUVS}>:cuvs::cuvs>)
+    target_link_libraries(swigfaiss_avx2 PRIVATE CUDA::cudart CUDA::cublas $<$<BOOL:${FAISS_ENABLE_CUVS}>:cuvs::cuvs>)
+    target_link_libraries(swigfaiss_avx512 PRIVATE CUDA::cudart CUDA::cublas $<$<BOOL:${FAISS_ENABLE_CUVS}>:cuvs::cuvs>)
+    target_link_libraries(swigfaiss_avx512_spr PRIVATE CUDA::cudart CUDA::cublas $<$<BOOL:${FAISS_ENABLE_CUVS}>:cuvs::cuvs>)
+    target_link_libraries(swigfaiss_sve PRIVATE CUDA::cudart CUDA::cublas $<$<BOOL:${FAISS_ENABLE_CUVS}>:cuvs::cuvs>)
   endif()
 endif()
 
